monitoring/rid_qualifier/mock/routes_behavior.py

```diff
@@ -0,0 +1,44 @@
+from typing import Tuple
+
+import flask
+
+from monitoring.monitorlib.typing import ImplicitDict
+from monitoring.rid_qualifier.mock import behavior
+from monitoring.rid_qualifier.mock.database import db
+from . import webapp
+
+
+@webapp.route('/dp/<dp_id>/behavior', methods=['PUT'])
+def set_dp_behavior(dp_id: str) -> Tuple[str, int]:
+  """Set the behavior of the specified virtual Display Provider."""
+  try:
+    json = flask.request.json
+    if json is None:
+      raise ValueError('Request did not contain a JSON payload')
+    dp_behavior = ImplicitDict.parse(json, behavior.DisplayProviderBehavior)
+  except ValueError as e:
+    msg = 'Change behavior for Display Provider {} unable to parse JSON: {}'.format(dp_id, e)
+    return msg, 400
+
+  dp = db.get_dp(dp_id)
```

Comment:
Recommend a try/catch to check for invalid dp/sp id passed.

Reply:
Getting the dp/sp [will always succeed](https://github.com/BenjaminPelletier/dss/blob/852aab84ff11de46c6c17fde9ee4c6c0209a0892/monitoring/rid_qualifier/mock/database.py#L51)